specification/changeanalysis/resource-manager/Microsoft.ChangeAnalysis/stable/2021-04-01/changeanalysis.json

```diff
@@ -0,0 +1,448 @@
+          "type": "string"
+        }
+      }
+    },
+    "ResourceProviderOperationDefinition": {
+      "description": "The resource provider operation definition.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The resource provider operation name.",
+          "type": "string"
+        },
+        "display": {
+          "$ref": "#/definitions/ResourceProviderOperationDisplay"
+        }
+      }
+    },
+    "ResourceProviderOperationList": {
+      "description": "The resource provider operation list.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Resource provider operations list.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceProviderOperationDefinition"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next page for list of Azure operations.",
+          "type": "string"
+        }
+      }
+    },
+    "ChangeType": {
+      "description": "The type of the change.",
+      "enum": [
+        "Add",
+        "Remove",
+        "Update"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ChangeType",
+        "modelAsString": true
+      }
+    },
+    "Level": {
+      "enum": [
+        "Noisy",
+        "Normal",
+        "Important"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "Level",
+        "modelAsString": true
+      }
+    },
+    "PropertyChange": {
+      "description": "Data of a property change.",
+      "type": "object",
+      "properties": {
+        "changeType": {
+          "$ref": "#/definitions/ChangeType"
+        },
+        "changeCategory": {
+          "description": "The change category.",
+          "enum": [
+            "User",
+            "System"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ChangeCategory",
+            "modelAsString": false
+          }
+        },
+        "jsonPath": {
+          "description": "The json path of the changed property.",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "The enhanced display name of the json path. E.g., the json path value[0].properties will be translated to something meaningful like slots[\"Staging\"].properties.",
+          "type": "string"
+        },
+        "level": {
+          "$ref": "#/definitions/Level"
+        },
+        "description": {
+          "description": "The description of the changed property.",
+          "type": "string"
+        },
+        "oldValue": {
+          "description": "The value of the property before the change.",
+          "type": "string"
+        },
+        "newValue": {
+          "description": "The value of the property after the change.",
+          "type": "string"
+        },
+        "isDataMasked": {
```

Comment:
As I explained in another comment, it's a GET API, but we set it as Unauthorized in ARM manifest. And we leverage the Microsoft.Authorization/checkAccess API to perform access check ourselves, so it's as reliable as ARM.

To be more specific, ARM check actionName/read for GET API and actionName/action for POST API, we are doing the same thing. i.e., if we get the data via POST API, we check if user has actionName/action permission, if we get the data from GET API, we check actionName/read permission.

So if the access check is reliable, can we use GET API or we still have to use POST?

Reply:
I think the only reason we use GET but not POST in the very beginning is because there is no request body. But sounds like no secret from GET API weigh much more than no request body of POST API? If you can confirm, I'll change the method to POST.
Also, just to confirm, we already have this GET API used by our partner with 2020-04-01-preview version, and it's OK that we have both GET and POST on the same API path correct?
I'm asking this because I see some RP create different APIs for secret vs non secret. E.g., GET /config vs POST /config/listSecrets. I wonder if it's technical reason they can't use same API path or they just choose to have a different API path.

Reply:
I suppose there is no problem having both API below and switched the method to POST.
GET /resourceChanges?api-version=2020-04-01-preview
POST /resourceChanges?api-version=2021-04-01

We will work with our partner team to deprecate the preview version.